src/Microsoft.Azure.Relay/Common/InputQueue.cs

```diff
@@ -603,7 +585,7 @@ public AsyncQueueReader(InputQueue<T> inputQueue, CancellationToken cancellation
                 : base(state)
             {
                 this.inputQueue = inputQueue;
-                this.cancelRegistration = cancellationToken.Register(s => CancelCallback(s), this);
+                this.cancelRegistration = cancellationToken.Register(CancelCallback, this);
```

Comment:
Undo this change for fewer allocations as mentioned above. (Yes I know VS2017 or some plugins try to correct this but I want fewer allocations, not slightly less code).